packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -353,7 +357,62 @@ module.exports = function s3 (config) {
     }, next)
   }
 
+  const policy = {
+    Version: '2012-10-17',
+    Statement: [
+      {
+        Effect: 'Allow',
+        Action: [
+          's3:PutObject',
+        ],
+        Resource: [
+          `arn:aws:s3:::${config.bucket}/*`,
+          `arn:aws:s3:::${config.bucket}`,
+        ],
+      },
+    ],
+  }
+
+  let stsClient
+  function getSTSClient () {
+    if (stsClient == null) {
+      stsClient = new STSClient({
+        region: config.region,
+        credentials : {
+          accessKeyId: config.key,
+          secretAccessKey: config.secret,
+        },
+      })
+    }
+    return stsClient
+  }
+
+  function getTemporarySecurityCredentials (req, res, next) {
```

Comment:
needs docs

Reply:
We don't document any of those endpoints on uppy.io, what do you have in mind?

Reply:
OK but what do you want to see documented exactly? I think this function has a fair amount of comments in it already, is there one specific part that should be clarified?

Reply:
below you explain the code but I'm missing _why_ this function exists. 

- why do we have temporary security credentials in companion?
- by what and when is this endpoint called?
- what are the S3 docs links to understand this concept?